pkg/operator/ceph/cluster/mgr/dashboard.go

```diff
@@ -94,92 +89,103 @@ func (c *Cluster) configureDashboard(m *mgrConfig) error {
+		changed, err := c.configureDashboardModuleSettings(daemonID)
+		if err != nil {
+			return err
+		}
+		if changed {
+			hasChanged = true
+		}
+	}
+	if hasChanged {
+		logger.Infof("dashboard config has changed. restarting the dashboard module.")
+		return c.restartDashboard()
+	}
 	return nil
 }
 
-func (c *Cluster) configureDashboardModule(m *mgrConfig) error {
+func (c *Cluster) configureDashboardModuleSettings(daemonID string) (bool, error) {
 	// url prefix
-	hasChanged, err := client.MgrSetConfig(c.context, c.Namespace, m.DaemonID, c.clusterInfo.CephVersion, "mgr/dashboard/url_prefix", c.dashboard.UrlPrefix, false)
+	hasChanged, err := client.MgrSetConfig(c.context, c.Namespace, daemonID, c.clusterInfo.CephVersion, "mgr/dashboard/url_prefix", c.dashboard.UrlPrefix, false)
 	if err != nil {
-		return err
+		return false, err
 	}
 
 	// ssl support
 	ssl := strconv.FormatBool(c.dashboard.SSL)
-	changed, err := client.MgrSetConfig(c.context, c.Namespace, m.DaemonID, c.clusterInfo.CephVersion, "mgr/dashboard/ssl", ssl, false)
+	changed, err := client.MgrSetConfig(c.context, c.Namespace, daemonID, c.clusterInfo.CephVersion, "mgr/dashboard/ssl", ssl, false)
 	if err != nil {
-		return err
+		return false, err
 	}
 	hasChanged = hasChanged || changed
 
 	// server port
-	port := strconv.Itoa(m.DashboardPort)
-	changed, err = client.MgrSetConfig(c.context, c.Namespace, m.DaemonID, c.clusterInfo.CephVersion, "mgr/dashboard/server_port", port, false)
+	port := strconv.Itoa(c.dashboardPort())
+	changed, err = client.MgrSetConfig(c.context, c.Namespace, daemonID, c.clusterInfo.CephVersion, "mgr/dashboard/server_port", port, false)
 	if err != nil {
-		return err
+		return false, err
 	}
 	hasChanged = hasChanged || changed
 
 	// SSL enabled. Needed to set specifically the ssl port setting starting with Nautilus(14.2.1)
 	if c.dashboard.SSL {
 		if c.clusterInfo.CephVersion.IsAtLeast(cephver.CephVersion{Major: 14, Minor: 2, Extra: 1}) {
-			changed, err = client.MgrSetConfig(c.context, c.Namespace, m.DaemonID, c.clusterInfo.CephVersion, "mgr/dashboard/ssl_server_port", port, false)
+			changed, err = client.MgrSetConfig(c.context, c.Namespace, daemonID, c.clusterInfo.CephVersion, "mgr/dashboard/ssl_server_port", port, false)
 			if err != nil {
-				return err
+				return false, err
 			}
 			hasChanged = hasChanged || changed
 		}
 	}
 
-	if hasChanged {
-		logger.Infof("dashboard config has changed")
-		return c.restartDashboard()
-	}
-	return nil
+	return hasChanged, nil
 }
 
-func (c *Cluster) initializeSecureDashboard() error {
+func (c *Cluster) initializeSecureDashboard() (bool, error) {
 	// we need to wait a short period after enabling the module before we can call the `ceph dashboard` commands.
 	time.Sleep(dashboardInitWaitTime)
 
 	password, err := c.getOrGenerateDashboardPassword()
 	if err != nil {
-		return fmt.Errorf("failed to generate a password. %+v", err)
+		return false, fmt.Errorf("failed to generate a password. %+v", err)
```

Comment:
```suggestion
		return false, fmt.Errorf("failed to generate a password for the ceph dashboard. %+v", err)
```

?

Reply:
The caller will actually append that this was for the dashboard, so i don't think this is needed.